python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -449,7 +450,26 @@ class RandomForestClassifier(BaseRandomForestModel,
         if self.random_state is None:
             seed_val = <uintptr_t>NULL
         else:
-            seed_val = <uintptr_t>self.random_state
+            if isinstance(self.random_state, np.uintp):
+                seed_val = <uintptr_t>self.random_state
+            else:
+                rs = self.random_state
+                if isinstance(rs, np.random.RandomState) or \
+                   isinstance(rs, cp.random.RandomState):
+                    seed_val = <uintptr_t>rs.randint(
+                        low=0,
+                        high=np.iinfo(np.uintp).max,
+                        dtype=np.uintp)
+                elif isinstance(rs, np.random.Generator):
+                    seed_val = <uintptr_t>rs.integers(
+                        low=0,
+                        high=np.iinfo(np.uintp).max,
+                        dtype=np.uintp)
+                else:
```

Comment:
Which type of `random_state` is this else condition mainly trying to tackle?

Reply:
When the random_state is not of type np.uintp (we use this because we later need to cast it into uintptr_t and this can directly be done), nor is the random_state a RandomState or Generator object. That is, it is of types like None/int/np.int32/np.int64. We sample from the Generator to ensure that we do not cast a number that is out of bounds for unitptr_t. Note that this else condition also handles cases when we supply a float/string/other forbiddened datatypes since the Generator into which our random_state object is fed as a seed will throw an error in such cases.